src/Sparrow/Json/BlittableJsonReaderVector.cs

```diff
@@ -0,0 +1,567 @@
+
+            _mem = mem;
+            _header = (BlittableVectorHeader*)mem; // +1 to skip the token byte
+            _dataStart = mem + sizeof(BlittableVectorHeader) + _header->AlignmentOffset;
+        }
+
+        public object this[int i]
+        {
+            get
+            {
+                switch (_header->Type)
+                {
+                    case BlittableVectorType.SByte:
+                        return *((sbyte*)_dataStart + i);
+                    case BlittableVectorType.Int16:
+                        return *((short*)_dataStart + i);
+                    case BlittableVectorType.Int32:
+                        return *((int*)_dataStart + i);
+                    case BlittableVectorType.Int64:
+                        return *((long*)_dataStart + i);
+                    case BlittableVectorType.Byte:
+                        return *((byte*)_dataStart + i);
+                    case BlittableVectorType.UInt16:
+                        return *((ushort*)_dataStart + i);
+                    case BlittableVectorType.UInt32:
+                        return *((uint*)_dataStart + i);
+                    case BlittableVectorType.UInt64:
+                        return *((ulong*)_dataStart + i);
+#if NET6_0_OR_GREATER
+                    case BlittableVectorType.Half:
+                        return *((Half*)_dataStart + i);
+#endif
+                    case BlittableVectorType.Float:
+                        return *((float*)_dataStart + i);
+                    case BlittableVectorType.Double:
+                        return *((double*)_dataStart + i);
+                }
+
+                throw new InvalidOperationException($"Vector is not of any known type {_header->Type}");
+            }
+        }
+
+        public bool IsTypeCompatibleForDirectRead<T>()
+        {
+            var type = typeof(T);
+            return _header->Type switch
+            {
+                BlittableVectorType.Byte or BlittableVectorType.SByte => type == typeof(sbyte) || type == typeof(byte),
+                BlittableVectorType.Int16 => type == typeof(short) || type == typeof(ushort),
+                BlittableVectorType.Int32 => type == typeof(int) || type == typeof(uint),
+                BlittableVectorType.Int64 => type == typeof(long) || type == typeof(ulong),
+                BlittableVectorType.UInt16 => type == typeof(ushort) || type == typeof(byte),
+                BlittableVectorType.UInt32 => type == typeof(uint) || type == typeof(byte),
+                BlittableVectorType.UInt64 => type == typeof(ulong) || type == typeof(byte),
+                BlittableVectorType.Float => type == typeof(float),
+                BlittableVectorType.Double => type == typeof(double),
+                #if NET6_0_OR_GREATER
+                BlittableVectorType.Half => type == typeof(Half),
+                #endif
+                _ => false
+            };
+
```

Comment:
`Int16` can be technically read as `UInt16 -> ushort` but then if we are allowing cohersion why not `UInt16 -> short`? Also if we are allowing narrowing we should support it at all levels.